Change the #warn for compiling using DM to #error so the compilor errors if you try to use DM to compile

MNarath1 opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

## About The Pull Request

This PR simply replaces the compile time warn directive with error so it stops compiling if you attempt to use DM to compile instead of just dropping a warning that most people appearantly ignore.
You might think this is obsolete but there have been multiple cases of people asking for help because their tgui chat didn't work just for it to turn out that they compiled using dm.

## Why It's Good For The Game

Instead of asking why their tgui chat isn't working people now will ask why they cannot compile with dm.

## Changelog
:cl:
tweak: make compiler error if someone tries to compile with Dream Maker
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

MNarath1 committed:
make compile error if you try to use DM

DatBoiTim commented:
Hehehehe CBT

Crossedfall commented:
Mixed feelings on this. I don't know if we should really hard block the ability to build through DM directly. That likely will have lasting ramifications.

MNarath1 commented:
> Mixed feelings on this. I don't know if we should really hard block the ability to build through DM directly. That likely will have lasting ramifications.

Well its in the end your call if it goes in or not i only know i had around 3 people in around 2 weeks that all tried to compile with DM and did not read the warning that immidiatly popped up (to be fair the warning goes under because of the log moving to fast) and then asked around why their chat didn't work with everyone going on a cat and mouse hunt for 20 minutes until finally it comes out that they simply didn't use the compile script

MNarath1 commented:
> > Mixed feelings on this. I don't know if we should really hard block the ability to build through DM directly. That likely will have lasting ramifications.
> 
> Well its in the end your call if it goes in or not i only know i had around 3 people in around 2 weeks that all tried to compile with DM and did not read the warning that immidiatly popped up (to be fair the warning goes under because of the log moving to fast) and then asked around why their chat didn't work with everyone going on a cat and mouse hunt for 20 minutes until finally it comes out that they simply didn't use the compile script

i wish the warning would at least be more obvious than just a small note in the compile log that immidiatly gets swepped away in its current state at least its extremly easy to miss

MNarath1 commented:
Alternativly i guess it would also work to make the fallback for TGUI (that displays when tgui chat doesn't load aka fancy chat) display a warning that this could be because it was compiled using DM

BrumalLake commented:
it should be made hard for the average person to break and people who want to do something strange can change it for themselves

MNarath1 commented:
@Crossedfall i tought about it and how about doing it like this I will keep it this way but add another commented out define that when uncommented in the file will supress the error message just like the CBT define would do and add an additional error directive explaining that you can still attempt to compile with dm by uncommenting said define

MNarath1 committed:
add a define that allows to cercumvent the compiler error

If someone really wants to compile using DM for some reason they just have to uncomment this define

MNarath1 committed:
comment change
